src/app/services/wiseLinkService.ts

```diff
@@ -7,42 +7,63 @@ import { UtilService } from '../../assets/wise5/services/utilService';
 @Injectable()
 export class WiseLinkService {
   constructor(
-    private NodeService: NodeService,
+    private nodeService: NodeService,
     private sanitizer: DomSanitizer,
-    private StudentDataService: StudentDataService,
-    private UtilService: UtilService
+    private studentDataService: StudentDataService,
+    private utilService: UtilService
   ) {}
 
-  createWiseLinkClickedHandler(currentNodeId: string): any {
+  wiseLinkClickedEventName: string = 'wiselinkclicked';
+  wiseLinkClickedHandler: any;
+  wiseLinkCommunicatorId: string = 'wise-link-communicator';
+
+  addWiseLinkClickedListener(): void {
+    this.wiseLinkClickedHandler = this.createWiseLinkClickedHandler();
+    this.getWiseLinkCommunicator().addEventListener(
+      this.wiseLinkClickedEventName,
+      this.wiseLinkClickedHandler
+    );
+  }
+
+  private createWiseLinkClickedHandler(): any {
+    const thisStudentDataService = this.studentDataService;
     return (event: CustomEvent) => {
+      const currentNodeId = thisStudentDataService.getCurrentNodeId();
       this.followLink(currentNodeId, event.detail.nodeId, event.detail.componentId);
     };
```

Comment:
Look up currentNodeId in the followLink function?
```suggestion
    return (event: CustomEvent) => {
      this.followLink(event.detail.nodeId, event.detail.componentId);
    };
```